### PR TITLE
Minor turret refactor

### LINKED ISSUE #17540: Crafted Turrets don't do shit to humans
Despite having like 5 settings for targeting crewmembers, none of them will aggro the turret. They appear to still target animals just fine though.

### LINKED ISSUE #18443: turrets don't work without a turret control computer
i'm not sure if those are even buildable, but at least it's deceptive; once the turrets are unlocked an AI can control it's target discrimination IFF and power state, however the turrets don't fire even if set to shoot at everything and the power is on. which is at least a little weird.

probably should put a note in the turret controls somewhere that says they won't fire without a control, or just make them default to a ready state without one.

### PR DESCRIPTION
**Pull request by AnturK**
Not much changes outwardly. Mostly code cleanup.
Emagging can't be turned off easily by humans now.

Fixes #17540
Fixes #18443

### CODE REVIEW

**Comment by SgtxRootbeer** [comment]
So if I'm a qm and I don't want as ash Walker uprising, turrets are finally a viable option.

**Comment by AnturK** [comment]
Actually need to modify this bit more.

**Comment by Cebasiz** [comment]
does this fix that problem where you can't actually build turrets because the device on the wall that toggles them on/off is not constructable(?) and they start in an off state regardless of whether or not they are set to "on" in the turret's control window thing?

**Comment by Cebasiz** [comment]
in fact if if it's in the scope of this thing can you just make a standalone mode for the turrets so they just shoot at anything that comes into view respecting settings and not bother with the wall-mounted on/off?

**Comment by SgtxRootbeer** [comment]
The wall device is a turret control frame and can be made in an autolathe, after being mounted on the wall, use a multi tool to link it.

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 1]
```diff
@@ -53,111 +62,117 @@
...
+	//optional properties
+	if(gun_properties["shot_delay"])
+		shot_delay = gun_properties["shot_delay"]
+	if(gun_properties["reqpower"])
+		reqpower = gun_properties["reqpower"]
+	
+	update_icon()
+	return gun_properties
+		
+		/*
```
Comment (optimumtact) [inline code/game/machinery/porta_turret/portable_turret.dm]: please delete this commented out code

**Comment by optimumtact** [comment]
Apart from my one comment, LGTM

File: code/game/machinery/porta_turret/portable_turret.dm (hunk 1 above)
Comment (AnturK) [inline code/game/machinery/porta_turret/portable_turret.dm]: Oh, whoops. Left this in to see if the sounds fit with oldcode. Will remove in a moment.

**Pull request merged**